test/scripts/upgrade-ovn.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# always exit on errors
+set -ex
+
+
+export KUBECONFIG=${HOME}/admin.conf
+export OVN_IMAGE=${OVN_IMAGE:-ovn-daemonset-f:pr}
+
+kubectl_wait_pods() {
+  # Check that everything is fine and running. IPv6 cluster seems to take a little
+  # longer to come up, so extend the wait time.
+  OVN_TIMEOUT=900s
+  if [ "$KIND_IPV6_SUPPORT" == true ]; then
+    OVN_TIMEOUT=1400s
+  fi
+  if ! kubectl wait -n ovn-kubernetes --for=condition=ready pods --all --timeout=${OVN_TIMEOUT} ; then
+    echo "some pods in OVN Kubernetes are not running"
+    kubectl get pods -A -o wide || true
+    kubectl describe po -n ovn-kubernetes
+    exit 1
+  fi
+  if ! kubectl wait -n kube-system --for=condition=ready pods --all --timeout=300s ; then
+    echo "some pods in the system are not running"
+    kubectl get pods -A -o wide || true
+    kubectl describe po -A
+    exit 1
+  fi
+}
+
+kubectl_wait_daemonset(){
+  # takes one daemonset and makes sure its desiredNumberScheduled and numberReady are equal
+  while true; do
+    run_kubectl get daemonsets.apps $1 -n ovn-kubernetes
+    DESIRED_REPLICAS=$(run_kubectl get daemonsets.apps $1 -n ovn-kubernetes -o=jsonpath='{.status.desiredNumberScheduled}')
+    READY_REPLICAS=$(run_kubectl get daemonsets.apps $1 -n ovn-kubernetes -o=jsonpath='{.status.numberReady}')
+    echo "CURRENT READY REPLICAS: $READY_REPLICAS, CURRENT DESIRED REPLICAS: $DESIRED_REPLICAS for the DaemonSet $1"
+    if [[ $READY_REPLICAS -eq $DESIRED_REPLICAS ]]; then
+      UP_TO_DATE_REPLICAS=$(run_kubectl get daemonsets.apps ovnkube-node -n ovn-kubernetes  -o=jsonpath='{.status.updatedNumberScheduled}')
+      echo "CURRENT UP TO DATE REPLICAS: $UP_TO_DATE_REPLICAS for the Deployment $1"
+      if [[ $READY_REPLICAS -eq $UP_TO_DATE_REPLICAS ]]; then
+        break
+      fi
+    fi
+    sleep 30
+  done
+  
+}
+
+kubectl_wait_deployment(){
+  # takes one deployment and makes sure its replicas and readyReplicas are equal
+  while true; do
```

Comment:
should there be some maximum tries here and then exit? otherwise this will run forever right?

Reply:
Yeah adding.